Bottom: 1a0b76da8b0ee860169899cf9053a50d0ec8b79d
Top:    a1a9068237fc924517c7506fcfcc6979a14ee630
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-01-22 17:16:30 +0800

Refresh of tuntap-xdp_tx-can-use-native

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index e0a7456..b24c830 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1675,7 +1675,7 @@ static struct sk_buff *tun_build_skb(struct tun_struct *tun,
 			tfile->xdp_flush_needed = true;
 			return NULL;
 		case XDP_TX:
-			err = tun_xdp_xmit(dev, xdp);
+			err = tun_xdp_xmit(tun->dev, &xdp);
 			if (err)
 				goto err_redirect;
 			rcu_read_unock();
